File: flow-data-service/src/test/java/com/bookislife/flow/data/BaseModifierTest.java
package com.bookislife.flow.data;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.ObjectWriter;
import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.*;

/**
 * Created by SidneyXu on 2016/06/12.
 */
public class BaseModifierTest {

    private ObjectWriter writer;

    @Before
    public void setup() {
        ObjectMapper objectMapper = new ObjectMapper();
        writer = objectMapper.writerWithDefaultPrettyPrinter();
    }

    @Test
    public void testModifier() throws Exception {
        BaseModifier modifier= BaseModifier.newBuilder()
                .addUpdater(BaseModifier.INC,"name",100)
                .create();


        writer.writeValue(System.out, modifier);
    }
}

File: flow-data-service/src/test/kotlin/com/bookislife/flow/data/BaseModifierTest.kt
package com.bookislife.flow.data

import org.junit.Test

/**
 * Created by SidneyXu on 2016/06/12.
 */
class BaseModifierTest : BaseTest() {

    @Test
    fun testModifier() {
        val modifier = BaseModifier.newBuilder()
                .addUpdater(BaseModifier.INC, "name", 100)
                .create()
        write(modifier)

        /*
            {
              "modifiers" : {
                "$inc" : {
                  "name" : 100
                }
              }
            }
         */
    }
}